_posts/blog/2017-07-19-google-assistant-development-for-android-devs-part-2.md

```diff
@@ -0,0 +1,107 @@
+---
+title: "Google Assistant Development for Android Devs (Part 2)"
+author: Patrick Jackson
+categories: Development
+tags: [Google Assistant, Actions on Google, conversational interfaces, Android, Kotlin]
+---
+
+
+# Google Assistant Development for Android Devs (Part 2)
+
+![](https://developers.google.com/actions/images/surfaces-hero.svg)
+
+This is part 2 in a series on developing Actions on Google.  [Part one is here](http://ticketmastermobilestudio.com/blog/google-assistant-development-for-android-devs-part-1).
+
+In my last post, I introduced an open source SDK for Actions on Google for Kotlin and Java.  Today, that library port is complete, with all features supported.  
+
+ - Api.ai support
+ - Actions SDK support
+ - All tests ported and passing (~230)
+ - Conversation components example for Api.Ai & Actions SDK
+ - Transaction Sample for Api.Ai
+ - Java conversation sample
+ 
+Now the capabilities of the [official Node.js SDK](https://github.com/TicketmasterMobileStudio/actions-on-google-kotlin) can be matched easily in Kotlin & Java (or any JVM language).  Since this is a port, and the API is nearly identical, one can look at the Node.js docs and examples and easily do the same in Kotlin or Java.  Below is how writing Actions for Google with the unofficial SDK looks.
+
+### Actions in Kotlin
+
+    fun welcome(app: ApiAiApp) =
```

Comment:
Let's stick to the backtick syntax for code blocks, since it allows us to specify language and isn't as prone to mistakes (your second block is not all 4-spaces indented, so it's throwing some errors right now):

    ```kotlin
    // blah blah
    ```

    ```java
    // blah blah
    ```